app/src/androidTest/java/com/android/sample/model/profile/MockProfilesRepository.kt

```diff
@@ -56,7 +56,13 @@ class MockProfilesRepository(private val database: MockUsersDatabase = MockUsers
   }
 
   override fun updateProfile(user: User, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
-    throw NotImplementedError("Not implemented: updateProfile")
+    try {
+      database.updateUser(user)
+
+      onSuccess()
+    } catch (e: Exception) {
+      onFailure(e)
+    }
```

Comment:
Thanks for providing this function. LGTM. Great that you did not forget to call onSuccess()